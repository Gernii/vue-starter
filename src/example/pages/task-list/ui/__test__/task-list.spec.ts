import { mount } from "@vue/test-utils";
import { createPinia, setActivePinia } from "pinia";
import { beforeEach, describe, expect, it } from "vitest";
import TaskList from "../task-list.vue";

describe("TaskList", () => {
	beforeEach(() => {
		// Set up a fresh Pinia instance
		setActivePinia(createPinia());
	});
	it("renders the task list properly", () => {
		const wrapper = mount(TaskList, {
			// Provide necessary props or slots if needed
		});

		// Add assertions based on the expected behavior of TaskList
		// Example:
		expect(wrapper.html()).toMatchSnapshot();
	});

	// Add more test cases as needed
});
