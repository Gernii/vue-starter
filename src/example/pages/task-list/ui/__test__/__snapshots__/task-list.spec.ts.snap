// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TaskList > renders the task list properly 1`] = `
"<button class="btn btn-primary btn-sm">Add <svg class="size-4 flex-shrink-0" viewBox="0 0 24 24" width="1.2em" height="1.2em">
    <path fill="none" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M5 12h14m-7-7v14"></path>
  </svg></button>
<!--teleport start-->
<!--teleport end-->"
`;
